src/main/java/com/exactpro/th2/crawler/util/CrawlerUtils.java

```diff
@@ -204,6 +199,59 @@ public static Iterator<StreamResponse> collectMessages(Iterator<StreamResponse>
                 messageData -> messageData.hasTimestamp() ? messageData.getTimestamp() : null);
     }
 
+    public static MessagesCrawlerData.ResumeMessageIDs updateResumeMessageIDs(MessagesCrawlerData.ResumeMessageIDs resumeMessageIDs,
+                                                                              Map<StreamKey, MessageID> startIds,
+                                                                              Map<StreamKey, MessageID> resumeIds) {
+        if (resumeMessageIDs != null) {
+            fillOppositeStream(resumeMessageIDs, startIds);
```

Comment:
I believe we should fill only the `startIDs`. The `resumeIDs` will be a subset of the `startIDs`. So, if we don't have the stream key in `resumeIDs` we will take the one from `startIDs`.

Also, we should fill the `startIDs` right where we collect them

Reply:
I don't really understand how to fill the `startIDs` where we collect them. I need to compare them with the previous ones. And as we collect `startIDs` in `requestData(...)` method, we don't don't have an access to the previous ones there.

Reply:
The algorithm I see:
1) We collect start IDs from the provider (the list of IDs from which we start)
2) If any of the session aliases requested from the provider does not have any direction we create one with -1 sequence for each missing direction
3) When we update the resume IDs we use start IDs as initial values and override the ones we received from the processor

@eugene-zheltov Do you have any objections to this?

Reply:
The problem is with the response from Processor, not with the response from Provider. 

If Provider doesn't find a messages of one of the streams, it returns the previous id of that stream (if `resumeId` is used in the request) or -1 (if the request uses only timestamp). So Provider will not cause this problem.

But it's possible that Processor doesn't return id for one of the streams. And for that case we need to fill it.

Reply:
Okay, the processor does not return some of the IDs. But they will present in start IDs, isn't it? Why can we take the value from start IDs in that case?